sdk/formrecognizer/Azure.AI.FormRecognizer/src/CopyModelOperation.cs

```diff
@@ -0,0 +1,213 @@
+             this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        /// <inheritdoc/>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) =>
+            UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <inheritdoc/>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) =>
+            await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Calls the server to get updated status of the long-running operation.
+        /// </summary>
+        /// <param name="async">When <c>true</c>, the method will be executed asynchronously; otherwise, it will execute synchronously.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>The HTTP response from the service.</returns>
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (!_hasCompleted)
+            {
+                Response<CopyOperationResult> update = async
+                    ? await _serviceClient.GetCustomModelCopyResultAsync(new Guid(_modelId), new Guid(_resultId), cancellationToken).ConfigureAwait(false)
+                    : _serviceClient.GetCustomModelCopyResult(new Guid(_modelId), new Guid(_resultId), cancellationToken);
+
+                _response = update.GetRawResponse();
+
+                if (update.Value.Status == OperationStatus.Succeeded)
+                {
+                    _hasCompleted = true;
+                    _value = ConvertValue(update.Value, _targetModelId, CustomFormModelStatus.Ready);
+                }
+                else if (update.Value.Status == OperationStatus.Failed)
+                {
+                    _hasCompleted = true;
+
+                    throw await CreateExceptionForFailedOperationAsync(async, update.Value.CopyResult.Errors).ConfigureAwait(false);
+                }
+            }
+
+            return GetRawResponse();
+        }
+
+        private static CustomFormModelInfo ConvertValue(CopyOperationResult result, string modelId, CustomFormModelStatus status)
+        {
+            return new CustomFormModelInfo(
+                modelId,
+                result.CreatedDateTime,
+                result.LastUpdatedDateTime,
+                status);
+        }
+
+        private async ValueTask<RequestFailedException> CreateExceptionForFailedOperationAsync(bool async, IReadOnlyList<FormRecognizerError> errors)
+        {
+            string errorMessage = default;
+            string errorCode = default;
+
+            if (errors.Count > 0)
+            {
+                var firstError = errors[0];
+
+                errorMessage = firstError.Message;
+                errorCode = firstError.ErrorCode;
+            }
+            else
```

Comment:
The only condition we use to consider it an error scenario is status being `OperationStatus.Failed`. If the service ever happens to return a failed status with no error listed (count = 0), we would still consider it an error.

Frankly speaking, I don't think we currently have any scenarios that are expected to reach this `else` block, but this could prevent us from throwing an `IndexOutOfRangeException` if the service behaves in an unexpected manner.